app/assets/javascripts/app/modules/personal-key-formatter.js

```diff
@@ -0,0 +1,45 @@
+import { DelimitedTextFormatter } from 'field-kit';
+
+/**
+ * @const
+ * @private
+ */
+const DIGITS_PATTERN = /^[a-zA-Z0-9]*$/;
+
+/**
+ * @extends DelimitedTextFormatter
+ */
+class PersonalKeyFormatter extends DelimitedTextFormatter {
+  constructor() {
+    super('-', true);
+    this.maximumLength = 16 + 3;
```

Comment:
maybe make these integers into constants just for readability? `PERSONAL_KEY_LENGTH` and `NUMBER_OF_DELIMITERS`